packages/splits-sdk/src/client/data.ts

```diff
@@ -0,0 +1,1061 @@
+      throw new AccountNotFoundError('split', splitAddress, chainId)
+
+    return await this.formatSplit(response)
+  }
+
+  protected async formatSplit(gqlSplit: ISplit): Promise<Split> {
+    const split = protectedFormatSplit(gqlSplit)
+
+    if (this._includeEnsNames) {
+      if (!this._ensPublicClient) throw new Error()
+      const ensRecipients = split.recipients
+        .map((recipient) => {
+          return recipient.recipient
+        })
+        .concat(split.controller ? [split.controller] : [])
+        .concat(
+          split.newPotentialController ? [split.newPotentialController] : [],
+        )
+
+      await addEnsNames(this._ensPublicClient, ensRecipients)
+    }
+
+    return split
+  }
+
+  async getAccountMetadata({
+    chainId,
+    accountAddress,
+  }: {
+    chainId: number
+    accountAddress: string
+  }): Promise<SplitsContract | undefined> {
+    validateAddress(accountAddress)
+    this._requirePublicClient()
+
+    const response = await this._loadAccount(accountAddress, chainId)
+
+    if (!response)
+      throw new AccountNotFoundError('account', accountAddress, chainId)
+
+    return await this._formatAccount(chainId, response)
+  }
+
+  // Helper functions
+  private async _formatAccount(
+    chainId: number,
+    gqlAccount: IAccountType,
+  ): Promise<SplitsContract | undefined> {
+    if (!gqlAccount) return
+
+    if (gqlAccount.type === 'split') return await this.formatSplit(gqlAccount)
+    else if (gqlAccount.type === 'waterfall')
+      return await this.formatWaterfallModule(chainId, gqlAccount)
+    else if (gqlAccount.type === 'liquidSplit')
+      return await this.formatLiquidSplit(gqlAccount)
+    else if (gqlAccount.type === 'swapper')
+      return await this.formatSwapper(gqlAccount)
+  }
+
+  // Graphql read actions
+  async getRelatedSplits({
+    chainId,
+    address,
+  }: {
+    chainId: number
+    address: string
+  }): Promise<{
+    receivingFrom: Split[]
+    controlling: Split[]
+    pendingControl: Split[]
+  }> {
+    validateAddress(address)
+
+    const response = await this._loadFullAccount(address, chainId)
+
+    const [receivingFrom, controlling, pendingControl] = await Promise.all([
+      Promise.all(
+        response.upstreamSplits
+          ? response.upstreamSplits.map(async (recipient) =>
+              this.formatSplit(recipient),
+            )
+          : [],
+      ),
+      Promise.all(
+        response.controllingSplits
+          ? response.controllingSplits.map(async (recipient) =>
+              this.formatSplit(recipient),
+            )
+          : [],
+      ),
+      Promise.all(
+        response.pendingControlSplits
+          ? response.pendingControlSplits.map(async (recipient) =>
+              this.formatSplit(recipient),
+            )
+          : [],
+      ),
+    ])
+
+    return {
+      receivingFrom,
+      controlling,
+      pendingControl,
+    }
```

Comment:
should we add v2 in here? will they already be included in `receivingFrom`?

dunno if it makes sense to lump v1 and v2 together or keep separate...

anyway, don't think that's a blocker for this pr. but probably something we should figure out.

Reply:
the controlling splits/upstream splits should include v2 as well automatically